### ISSUE TITLE
NPE @ PasteToWorld

### ISSUE THREAD

**Issue by mazetar** [tehbeard/LibSchematic#6]
Hi!

I'm just playing around with the lib, trying to get familiar with it and I got an NPE upon trying to paste from a schematic into the world in some chunks. 

The first time I tried to load a schematic it worked perfectly, then when I moved into some other chunk's I crashed due to an NPE. If I returned to the first chunk I found that I could load schematics there, without crashing.

I believe I found where it happens but before that, heres my crash log:
http://paste.minecraftforge.net/view/82692348
I have a block (BlockSchematic) which on activation simply loads the schematic.
I believe the implementation of it to be unimportant, as it's a ripoff from your LibSchematic.sh code.

The crash happens on line 38 of PasteToWorld.java
it does not happen at once, it seems to run a few iterations before crashing. It might just be in this chunk and different in others but in my current chunk it's crashing on the 64th pass of line 38.

The reason for the crash is as far as I can tell that storageArray is null at the time.
Indicating that chunk.getBlockStorageArray() returned null on line 36.

**Comment by mazetar** [comment]
I played around with it for a few minutes and got it working, but there are probably better ways to do this.

``` Java
if (storageArray == null) {
                ExtendedBlockStorage[] storageArrays = chunk.getBlockStorageArray();
                storageArrays[worldVector.getY() >> 4] = new ExtendedBlockStorage(storageArrays[(worldVector.getY() >> 4) - 1].getYLocation() + 16 ,!world.provider.hasNoSky);
                storageArray = chunk.getBlockStorageArray()[worldVector.getY() >> 4];
            }
```

**Comment by mazetar** [comment]
Fixed by 1d3a30d.
